src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -568,5 +580,17 @@ private void CheckSize(int countToAdd)
                 }
             }
         }
+
+        private struct EncodingPreamblePair
```

Comment:
@mikedn, you're probably right, but I'm not sure what's wrong with my microbenchmarking then.  Here's my example:
```C#
using System;
using System.Diagnostics;
using System.Linq;

static class Test
{
    sealed class WrapperClass
    {
        public object Object1;
        public object Object2;
    }

    struct WrapperStruct
    {
        public object Object1;
        public object Object2;
    }

    static void Main()
    {
        var classArr = Enumerable.Range(0, 10).Select(_ => new WrapperClass { Object1 = new object(), Object2 = new object() }).ToArray();
        var structArr = Enumerable.Range(0, 10).Select(_ => new WrapperStruct { Object1 = new object(), Object2 = new object() }).ToArray();
        var sw = new Stopwatch();
        const int Iters = 100000000;

        while (true)
        {
            sw.Restart();
            for (int i = 0; i < Iters; i++)
            {
                foreach (var item in structArr)
                {
                    object obj1 = item.Object1;
                    object obj2 = item.Object2;
                }
            }
            Console.WriteLine("Struct : " + sw.ElapsedMilliseconds);

            sw.Restart();
            for (int i = 0; i < Iters; i++)
            {
                foreach (var item in classArr)
                {
                    object obj1 = item.Object1;
                    object obj2 = item.Object2;
                }
            }
            Console.WriteLine("Class  : " + sw.ElapsedMilliseconds);

            Console.WriteLine();
        }
    }
}
```
With VS2015 RTM, on my machine the 32-bit JIT results in the class being 3x the speed of the struct version, and the 64-bit JIT results in it being 1.25x faster.  Do you see different results?